packages/compile-common/src/types.ts

```diff
@@ -0,0 +1,49 @@
+import { ContractObject } from "@truffle/contract-schema/spec";
+
+export type Compilation = {
+  sourceIndexes: string[];
+  contracts: CompiledContract[];
+  compiler: {
+    name: string | undefined;
+    version: string | undefined;
+  };
+};
+
+export type CompilerResult = Compilation[];
```

Comment:
Do you mean 
```
export interface CompilerResult {
  compilations: Compilation[];
}
``` ? 

So that would mean we would need to once again change the return value to look something like `{ compilations: [{}, {}] }`?

I would probably side with the half of you that says this is too much extra engineering...especially for an unknown abstract benefit. If we want, we can always alter `Compilation`. But maybe you had another case in mind that wouldn't be solved by this?

Reply:
Well, if, in the future, we want to return something besides `compilations`, then we'd have room to add it. I can't think of an example right now, though.

Reply:
Well I suppose I don't mind editing it, it just makes packing/unpacking a pain. Should we go ahead and update it?

Reply:
Let's do it. Just thought of another reason in favor of `interface CompilerResult`: we can add `contracts` to it (now or later) and then use `CompilerResult` for both compile-solidity,etc. and workflow-compile.